kubos/override/reimplemented_modules/cmakegen.py

```diff
@@ -0,0 +1,48 @@
+# Kubos SDK
+# Copyright (C) 2017 Kubos Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from yotta.lib.cmakegen import *
+
+class KubosCMakeGen(CMakeGen):
+    def _kCheckImmediateDeps(self, deps):
+        # make a copy so we don't recurse
+        new_deps = {}
+        dep_items = [i for i in deps.items()]
+        for name, component in dep_items:
+            if component.isTestDependency(): continue
+            if component.getLinkWholeArchive():
+                # surround 'whole archive' libs with the proper linker args
+                new_key = ('${YOTTA_LINK_WHOLE_ARCHIVE_PRE} ' \
```

Comment:
they are defined by the `kubos-gcc` target as part of the bigger pull request https://github.com/kubostech/kubos/pull/34/files#diff-49792d44a77fc658e7b4d737058dd6c4